osquery/sql/sqlite_encoding.cpp

```diff
@@ -31,29 +37,47 @@ static void b64SqliteValue(sqlite3_context* ctx,
 
   std::string input((char*)sqlite3_value_text(argv[0]));
   std::string result;
-  if (encode) {
-    result = base64Encode(input);
-  } else {
+  if (encode & B64_ENCODE) {
+    if ((encode & B64_ENCODE_CONDITIONAL) && !isPrintable(input)) {
+      result = base64Encode(input);
+    } else {
+      result = input;
```

Comment:
Yeah I think you're right, I wanted to do the boolean logic because it made sense during an intermediate stage of coding.